### PR TITLE
Improve generated cmake files verification and update lint.sh

### PR DESCRIPTION
**Pull request by pzread**
- Add messages in the lint workflow to help fix the issue.
- Update `lint.sh`.

Sample failure: https://github.com/iree-org/iree/actions/runs/3588087186/jobs/6039094068

### CODE REVIEW

File: build_tools/github_actions/configure_ci.py [hunk 1]
```diff
@@ -57,9 +57,8 @@ def skip_path(path: str) -> bool:
 
 def set_output(d: Mapping[str, str]):
   print(f"Setting outputs: {d}")
-  step_output_file = os.environ["GITHUB_OUTPUT"]
-  with open(step_output_file, "a") as f:
-    f.writelines(f"{k}={v}" "\n" for k, v in d.items())
+  for k, v in d.items():
```
Comment (GMNGeoffrey) [inline build_tools/github_actions/configure_ci.py:60]: Please don't undo this :-)

Comment (pzread) [reply build_tools/github_actions/configure_ci.py]: Oh, files got messed up when testing the linter...

File: build_tools/scripts/generate_cmake_files.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ROOT_DIR=$(git rev-parse --show-toplevel)
+cd "${ROOT_DIR}"
+
+# Generates E2E test artifacts cmake files
+./build_tools/testing/generate_cmake_e2e_test_artifacts_suite.py \
+  --output_dir ./tests/e2e/test_artifacts
+
+# Generates E2E model test cmake files
+./build_tools/testing/generate_cmake_e2e_model_tests.py \
+  --output ./tests/e2e/models/generated_e2e_model_tests.cmake
```
Comment (ScottTodd) [inline build_tools/scripts/generate_cmake_files.sh:22]: nit: these comments are redundant IMO (the script names say the same thing)

Also, are the output paths ever different? You could just hardcode the paths into the scripts themselves - less places to update / keep in sync

**Review by ScottTodd — APPROVED** [review]

File: build_tools/scripts/generate_cmake_files.sh (hunk 2 above)
Comment (pzread) [reply build_tools/scripts/generate_cmake_files.sh]: Removed the comments.

IMO passing paths through command lines makes it easier to understand where those files are created. These paths are also relative to the git root. It's clearer to get git root and change the directory in shell script then in Python.

I updated the related READMEs so now all places point to this script to regenerate the cmake files. So this script is the only place to have these paths.

**Pull request merged**